### PR TITLE
refactor: replace 'Fn' type with function type expression (closes #565)

### LINKED ISSUE #565: Replace Fn type with function with named arguments
We need to replace `Fn` type in all code by a normal function type - it will be force to use names for arguments.

Do you have interest as a contributor? Add a comment of which package you want to refactor and come with PR lately!

### PR DESCRIPTION
**Pull request by alexvyber**
closes #565

### CODE REVIEW

**Comment by artalar** [comment]
Hi! Thank you for this, it is a lot of work, I appreciate it! I need a few days to review the changes and add some suggestions around arguments namenes.

File: docs/src/pages/guides/custom-operator.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Operator for the pipe function should starts with a verb.
 import { Atom, CtxSpy } from '@reatom/core'
 
 declare function mapState<T, Res>(
-  mapper: Fn<[CtxSpy, T, undefined | T], Res>,
+  mapper: (ctx: CtxSpy, arg1: T,  arg2: undefined | T) => Res,
```
Comment (artalar) [inline docs/src/pages/guides/custom-operator.md:19]:
```suggestion
  mapper: (ctx: CtxSpy, state: T,  prevState: undefined | T) => Res,
```

File: docs/src/pages/packages/testing.md [hunk 2]
```diff
@@ -16,9 +16,9 @@ import { createTestCtx, mockFn } from '@reatom/testing'
 export interface TestCtx extends Ctx {
   mock<T>(anAtom: Atom<T>, fallback: T): Unsubscribe
 
-  mockAction<T>(anAction: Action<any[], T>, cb: Fn<[Ctx], T>): Unsubscribe
+  mockAction<T>(anAction: Action<any[], T>, cb: (ctx: Ctx) => T): Unsubscribe
 
-  subscribeTrack<T, F extends Fn<[T]>>(
+  subscribeTrack<T, F extends (arg: T) => any>(
```
Comment (artalar) [inline docs/src/pages/packages/testing.md:29]:
```suggestion
  subscribeTrack<T, F extends (state: T) => any>(
```

File: packages/async/src/index.ts [hunk 3]
```diff
@@ -47,10 +46,14 @@ export interface AsyncCtx extends Ctx {
...
-  onFulfill?: Fn<[Ctx, Resp]>
-  onReject?: Fn<[Ctx, unknown]>
-  onSettle?: Fn<[Ctx]>
+  onEffect?: (
+    ctx: Ctx,
+    params: Params,
+    controlledPromise: ControlledPromise<Resp>,
+  ) => any
+  onFulfill?: (ctx: Ctx, resp: Resp) => any
+  onReject?: (ctx: Ctx, arg: unknown) => any
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
  onReject?: (ctx: Ctx, error: unknown) => any
```

File: packages/async/src/index.ts [hunk 4]
```diff
@@ -171,57 +175,55 @@ export const withDataAtom: {
     T extends AsyncAction & {
       dataAtom?: AsyncDataAtom<ActionPayload<T['onFulfill']>>
     },
-  >(): Fn<
-    [T],
-    T & { dataAtom: AtomMut<undefined | ActionPayload<T['onFulfill']>> }
-  >
+  >(): (
+    arg: T,
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
    anAsync: T,
```

File: packages/async/src/index.ts [hunk 5]
```diff
@@ -171,57 +175,55 @@ export const withDataAtom: {
...
+  ) => T & { dataAtom: AtomMut<undefined | ActionPayload<T['onFulfill']>> }
+
   <
     T extends AsyncAction & {
       dataAtom?: AsyncDataAtom<ActionPayload<T['onFulfill']>>
     },
   >(
     initState: ActionPayload<T['onFulfill']>,
-  ): Fn<[T], T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }>
+  ): (arg: T) => T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
  ): (anAsync: T) => T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }
```

File: packages/async/src/index.ts [hunk 6]
```diff
@@ -171,57 +175,55 @@ export const withDataAtom: {
...
     },
     State,
   >(
     initState: State,
-  ): Fn<
-    [T],
-    T & { dataAtom: AsyncDataAtom<State | ActionPayload<T['onFulfill']>> }
-  >
+  ): (
+    arg: T,
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
    anAsync: T,
```

File: packages/async/src/index.ts [hunk 7]
```diff
@@ -171,57 +175,55 @@ export const withDataAtom: {
...
-      ],
-      ActionPayload<T['onFulfill']>
-    >,
-  ): Fn<[T], T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }>
+    map?: (
+      ctx: Ctx,
+      payload: ActionPayload<T['onFulfill']>,
+      state: ActionPayload<T['onFulfill']>,
+    ) => ActionPayload<T['onFulfill']>,
+  ): (arg: T) => T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
  ): (anAsync: T) => T & { dataAtom: AsyncDataAtom<ActionPayload<T['onFulfill']>> }
```

File: packages/async/src/index.ts [hunk 8]
```diff
@@ -171,57 +175,55 @@ export const withDataAtom: {
...
-      [ctx: Ctx, payload: ActionPayload<T['onFulfill']>, state: State],
-      State
-    >,
-  ): Fn<[T], T & { dataAtom: AsyncDataAtom<State> }>
+    map?: (
+      ctx: Ctx,
+      payload: ActionPayload<T['onFulfill']>,
+      state: State,
+    ) => State,
+  ): (arg: T) => T & { dataAtom: AsyncDataAtom<State> }
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
  ): (anAsync: T) => T & { dataAtom: AsyncDataAtom<State> }
```

File: packages/async/src/index.ts [hunk 9]
```diff
@@ -250,7 +252,7 @@ export const withErrorAtom =
     },
     Err = Error,
   >(
-    parseError: Fn<[Ctx, unknown], Err> = (ctx, e) =>
+    parseError: (ctx: Ctx, arg: unknown) => Err = (ctx, e) =>
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
    parseError: (ctx: Ctx, error: unknown) => Err = (ctx, e) =>
```

File: packages/async/src/index.ts [hunk 10]
```diff
@@ -263,7 +265,9 @@ export const withErrorAtom =
     } = {
       resetTrigger: 'onEffect',
     },
-  ): Fn<[T], T & { errorAtom: Atom<undefined | Err> & { reset: Action } }> =>
+  ): ((
+    arg: T,
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
    anAsync: T,
```

File: packages/async/src/index.ts [hunk 11]
```diff
@@ -299,14 +303,11 @@ export const withAbort =
...
     strategy = 'last-in-win',
-  }: { strategy?: 'none' | 'last-in-win' } = {}): Fn<
-    [T],
-    T & {
-      abort: Action<[reason?: string], void>
-      onAbort: Action<[Error], Error>
-      abortControllerAtom: Atom<AbortController | null>
-    }
-  > =>
+  }: { strategy?: 'none' | 'last-in-win' } = {}): ((arg: T) => T & {
```
Comment (artalar) [inline packages/async/src/index.ts]:
```suggestion
  }: { strategy?: 'none' | 'last-in-win' } = {}): ((anAsync: T) => T & {
```

File: packages/async/src/withCache.ts [hunk 12]
```diff
@@ -82,19 +82,16 @@ export const withCache =
...
   // @ts-ignore
   (anAsync) => {
     if (!anAsync.cacheAtom) {
-      const find: Fn<
-        [Ctx, any[]],
-        { cached: CacheMapRecord<T>; key: unknown }
-      > = paramsToKey
+      const find: (
+        ctx: Ctx,
+        arg: any[],
```
Comment (artalar) [inline packages/async/src/withCache.ts]:
```suggestion
        params: any[],
```

**Review by artalar — CHANGES_REQUESTED** [review]

**Comment by artalar** [comment]
@alexvyber please, merge the fresh changes from `v3`

**Comment by artalar** [comment]
@alexvyber could you merge from v3 one more time please?) I plan to publish the changes today

**Comment by alexvyber** [comment]
> @alexvyber could you merge from v3 one more time please?) I plan to publish the changes today

Whoops...
Sorry, I just this comment.
Today will do it.

**Comment by artalar** [comment]
Thank you :)

**Review by artalar — APPROVED** [review]

**Pull request merged**